Networking/NetworkingTests/Requests/DotcomRequestTests.swift

```diff
@@ -91,6 +95,78 @@ class DotcomRequestTests: XCTestCase {
+                                        method: .post,
+                                        path: sampleRPC,
+                                        parameters: sampleParameters,
+                                        availableAsRESTRequest: true)
+
+        // When
+        let output = try XCTUnwrap(request.asRESTRequest(with: sampleSiteAddress))
+
+        // Then
+        XCTAssertEqual(output.apiVersionPath, WooAPIVersion.none.path)
+        XCTAssertEqual(output.method, .post)
+        XCTAssertEqual(output.path, WordPressAPIVersion.wpMark2.path + sampleRPC)
+        let params = try XCTUnwrap(output.parameters as? [String: String])
+        XCTAssertEqual(params, sampleParameters)
+        XCTAssertEqual(output.siteURL, sampleSiteAddress)
+    }
+
+    func test_initializing_RESTRequest_throws_for_non_WPOrg_endpoints__even_if_availableAsRESTRequest_is_true() throws {
+        let apis = WordPressAPIVersion.allCases.filter({ $0.isWPOrgEndpoint == false })
+        for api in apis {
+            do {
+                let _ = try DotcomRequest(wordpressApiVersion: api,
+                                          method: .post,
+                                          path: sampleRPC,
+                                          parameters: sampleParameters,
+                                          availableAsRESTRequest: true)
+            } catch DotcomRequestError.apiVersionCannotBeAccessedUsingRESTAPI {
+                // Catch the expected error
+            } catch {
+                XCTFail("Unexpected error!")
+            }
+        }
+    }
+
+    func test_converting_into_RESTRequest_is_nil_when_availableAsRESTRequest_is_false() throws {
+        // Given
+        let request = try DotcomRequest(wordpressApiVersion: .wpMark2,
+                                        method: .post,
+                                        path: sampleRPC,
+                                        parameters: sampleParameters,
+                                        availableAsRESTRequest: false)
+
+        // Then
+        XCTAssertNil(request.asRESTRequest(with: sampleSiteAddress))
+    }
+
+    func test_the_converted_RESTRequest_path_does_not_have_sites_component() throws {
+        // Given
+        let pathWithSite = "/sites/12345/media/"
+        let request = try DotcomRequest(wordpressApiVersion: .wpMark2,
+                                        method: .post,
+                                        path: pathWithSite,
+                                        parameters: sampleParameters,
+                                        availableAsRESTRequest: true)
+
+        // When
+        let output = try XCTUnwrap(request.asRESTRequest(with: sampleSiteAddress))
+
+        // Then
+        XCTAssertEqual(output.apiVersionPath, WooAPIVersion.none.path)
+        XCTAssertEqual(output.method, .post)
+        XCTAssertEqual(output.path, WordPressAPIVersion.wpMark2.path + "media/")
```

Comment:
nit: maybe we can use the raw string for the expected path just so that it's easier to see how the output URL looks like

Reply:
Good point. Done in ed377d06273d3bf84d7952cfc11b7ff24ab1864c